projects/bp-gallery/src/views/gallery/search/helpers/search-filters.ts

```diff
@@ -0,0 +1,234 @@
+    if (year < 100) {
+      startTime = year === 49 ? '1900-01-01T00:00:00Z' : `19${year}-01-01T00:00:00Z`;
+      endTime = `19${year}-12-31T23:59:59Z`;
+    }
+
+    const time_range_tag_filter = {
+      start: {
+        gte: startTime,
+      },
+      end: {
+        lte: endTime,
+      },
+    };
+
+    return time_range_tag_filter;
+  }
+};
+
+const searchDecade = (searchParams: URLSearchParams, filters: PictureFiltersInput) => {
+  const timeParams = searchParams.getAll(SearchType.DECADE);
+  timeParams.forEach(timeParam => {
+    const time_range_tag_filter = paramToTimefilter(timeParam);
+    filters.and?.push({
+      or: [
+        {
+          time_range_tag: time_range_tag_filter,
+        },
+        {
+          verified_time_range_tag: time_range_tag_filter,
+        },
+      ],
+    });
+  });
+};
+
+const searchKeyword = (searchParams: URLSearchParams, filters: PictureFiltersInput) => {
+  const keywords = searchParams.getAll(SearchType.KEYWORD).map(decodeURIComponent);
+  keywords.forEach((keyword: string) => {
+    const keyword_tag_filter = {
+      name: {
+        containsi: keyword,
+      },
+    };
+    filters.and?.push({
+      or: [
+        {
+          keyword_tags: keyword_tag_filter,
+        },
+        {
+          verified_keyword_tags: keyword_tag_filter,
+        },
+      ],
+    });
+  });
+};
+
+const searchLocation = (searchParams: URLSearchParams, filters: PictureFiltersInput) => {
+  const locations = searchParams.getAll(SearchType.LOCATION).map(decodeURIComponent);
+  locations.forEach((location: string) => {
+    const location_tag_filter = {
+      name: {
+        containsi: location,
+      },
+    };
+    filters.and?.push({
+      or: [
+        {
+          location_tags: location_tag_filter,
+        },
+        {
+          verified_location_tags: location_tag_filter,
+        },
+      ],
+    });
+  });
+};
+
+const searchPerson = (searchParams: URLSearchParams, filters: PictureFiltersInput) => {
+  const persons = searchParams.getAll(SearchType.PERSON).map(decodeURIComponent);
+  persons.forEach((person: string) => {
+    const person_tag_filter = {
+      name: {
+        containsi: person,
+      },
+    };
+    filters.and?.push({
+      or: [
+        {
+          person_tags: person_tag_filter,
+        },
+        {
+          verified_person_tags: person_tag_filter,
+        },
+      ],
+    });
+  });
+};
+
+const searchDescription = (searchParams: URLSearchParams, filters: PictureFiltersInput) => {
+  const q = searchParams.getAll(SearchType.DESCRIPTION).map(decodeURIComponent);
+  q.forEach((param: string) => {
+    filters.and?.push({
+      descriptions: {
+        text: {
+          containsi: param,
+        },
+      },
+    });
+  });
+};
+
+const searchAll = (searchParams: URLSearchParams, filters: PictureFiltersInput) => {
+  const params = searchParams.getAll(SearchType.ALL).map(decodeURIComponent);
+  params.forEach(function (value) {
```

Comment:
Is there a reason you used the `function` keyword here (and not the arrow syntax how in most places)? :D